README.md

```diff
@@ -2,7 +2,141 @@
 
 [![Build Status](https://travis-ci.org/lesfurets/pipeline-test-helper.svg?branch=develop)](https://travis-ci.org/lesfurets/pipeline-test-helper)
 
+Helpers for testing Jenkins pipeline code, written in [Groovy Pipeline DSL](https://jenkins.io/doc/book/pipeline/).
+
+If you use Jenkins as your CI workhorse (like us @[lesfurets.com](https://www.lesfurets.com)) and you enjoy writing _pipeline-as-code_, 
+you know already that pipeline code can get pretty complex.
+
+These helpers lets you write unit tests on the configuration and conditional logic of the pipeline code, by providing a mock execution of the pipeline.
+You can mock built-in Jenkins commands, job configurations, see the stacktrace of the whole execution and even track regressions.
 
 ## Usage
 
-## Writing Pipeline Tests
\ No newline at end of file
+### Add to your project as test dependency
+
+Maven: 
+
+```xml
+    <dependency>
+      <groupId>com.lesfurets</groupId>
+      <artifactId>pipeline-test-helper</artifactId>
+      <version>0.10</version>
+      <scope>test</scope>
+    </dependency>
+```
+
+Gradle:
+
+```groovy
+testCompile group:'com.lesfurets', name:'pipeline-test-helper', version:'0.10'
+```
+
+### Start writing tests
+
+You can write your tests in Java or Groovy, using the test framework you prefer.
+Easiest entry point is extending the abstract class `BasePipelineTest`.
+
+```groovy
+
+class TestExampleJob extends BasePipelineTest {
+
+        @Test
+        void should_execute_without_errors() throws Exception {
+            def script = loadScript("job/exampleJob.jenkins")
+            script.execute()
+            printCallStack()
+        }
+}
+
+```
+
+### Mock Jenkins commands
+
+You can register interceptors to mock Jenkins commands, which may or may not return a result.
+
+```groovy
+    @Override
+    @Before
+    void setUp() throws Exception {
+        super.setUp()
+        helper.registerAllowedMethod("timeout", [Map.class, Closure.class], null)
+        helper.registerAllowedMethod(method("readFile", String.class), { file ->
+            return Files.contentOf(new File(file), Charset.forName("UTF-8"))
+        })
+    }
+```
+
+The test helper already includes some mocks, but the list is far from complete. 
+You need to _register allowed methods_ if you want to override these mock and add others. 
+Note that you need to provide a method signature and a callback (closure or lambda) in order to allow a method.
```

Comment:
It is called `allow` cause 
> Any method call which is not recognized will throw an exception.

But we could choose to call these methods `mock`, or `when` later on.